src/components/inputs/SearchSelect.jsx

```diff
@@ -0,0 +1,427 @@
+  },
+
+  _closeAllGroups() {
+    this.setState({ openGroupsId: [] });
+  },
+
+  //State Helpers: filterValue
+
+  _updateFilterValue(newFilterValue) {
+    this.setState({filterValue: newFilterValue});
+    this._suggestFirstOption(newFilterValue);
+  },
+
+  _clearFilterValue() {
+    this._updateFilterValue('');
+  },
+
+  //State Helpers: suggestedOption
+
+  _suggestFirstOption(filterValue) {
+    let suggestion = this._getOptionsIterator(filterValue)[0];
+    this._setSuggestedOption(suggestion);
+  },
+
+  _suggestPreviousOption(filterValue, suggestedOption) {
+    let optionsIterator = this._getOptionsIterator(filterValue);
+
+    let currentSuggestionIndex = suggestedOption ? _.findIndex(optionsIterator, (option) => option.id === suggestedOption.id) : -1;
+    let previousSuggestionIndex = currentSuggestionIndex > 0 ? currentSuggestionIndex - 1 : currentSuggestionIndex;
+
+    this._setSuggestedOption(optionsIterator[previousSuggestionIndex]);
+  },
+
+  _suggestNextOption(filterValue, suggestedOption) {
+    let optionsIterator = this._getOptionsIterator(filterValue);
+
+    let currentSuggestionIndex = suggestedOption ? _.findIndex(optionsIterator, (option) => option.id === suggestedOption.id) : -1;
+    let nextSuggestionIndex = currentSuggestionIndex < (optionsIterator.length - 1) ? currentSuggestionIndex + 1 : currentSuggestionIndex;
+
+    this._setSuggestedOption(optionsIterator[nextSuggestionIndex]);
+  },
+
+  _setSuggestedOption(option) {
+    this.setState({suggestedOption: option});
+    this._openSuggestedGroup(option);
+  },
+
+  _clearSuggestionOption() {
+    this._setSuggestedOption(null);
+  },
+
+  //State Helpers: filteredOptions
+
+  _getOptionsIterator(filterValue) {
+    let options = this._getFilteredOptions(filterValue);
+    return _.flatten(_.map(options, (option) => option.isGroup ? option.options : option));
+  },
+
+  _getFilteredOptions(filterValue) {
+    let {options, filterOption, hideGroupsWithNoMatch} = this.props;
+
+    //Filter grouped options
+    options = _.map(options, (group) => {
+      if (!group.isGroup) {
+        return group;
+      }
+      return {
+        ...group,
+        options: _.filter(group.options, (option) => !filterValue || filterOption(filterValue, option, group)),
+      };
+    });
+
+    //Remove empty groups if hideGroupsWithNoMatch == true
+    options = _.filter(options, (option) => !(option.isGroup && option.options.length === 0 && hideGroupsWithNoMatch));
+
+    //Filter non grouped options
+    options = _.filter(options, (option) => option.isGroup || (!filterValue || filterOption(filterValue, option)));
+
+    return options;
+  },
+
+  //Element Helpers
+
+  _openList() {
+    this.setState({isFocused: true});
+  },
+
+  _closeList() {
+    this._blurFilterInput();
+    this.setState({isFocused: false});
+    this._closeAllGroups();
+  },
+
+  _focusFilterInput() {
+    this.refs.searchInput.getDOMNode().focus();
```

Comment:
Also, instead of having `blurFilterInput` and `focusFilterInput` methods, you should compare `this.state.focused` and `prevState.focused` in `componentDidUpdate` and focus/blur the input there.